app/models/game.rb

```diff
@@ -35,7 +35,11 @@ def populate_board!
   end
 
   def render_piece(row, col)
-    piece = pieces.where('row_coordinate = ? AND column_coordinate = ?', row, col)
-    piece.empty? ? '' : piece.first.color.capitalize! + ' ' + piece.first.type
+    piece = piece_at_location(row, col)
+    piece_at_location(row, col).empty? ? '' : piece.first.color.capitalize! + ' ' + piece.first.type
+  end
+
+  def piece_at_location(row, col)
+    pieces.where('row_coordinate = ? AND column_coordinate = ?', row, col)
```

Comment:
Could also write this as

`pieces.find_by('row_coordinate = ? AND column_coordinate = ?', row, col)`

Then you could remove the calls to `first` in `render_piece`